File: Templates/TypeScript/src/db/InMemoryDatabase.ts
import { IDatabase } from "./IDatabase";
import { Student } from "../models/Student";

export class InMemoryDatabase implements IDatabase {
    private students = new Map<number, Student>();

    createStudent(student: Student): Student {
        // Find the maximum current id and increment it for the new student
        const nextId = this.students.size > 0
            ? Math.max(...Array.from(this.students.keys())) + 1
            : 0;
        student.id = nextId; // Assign the new id
        this.students.set(student.id, student);
        return student;
    }

    getStudent(id: number): Student | undefined {
        return this.students.get(id);
    }

    updateStudent(student: Student): void {
        this.students.set(student.id, student);
    }

    deleteStudent(id: number): void {
        this.students.delete(id);
    }

    getAllStudents(): Student[] {
        return Array.from(this.students.values());
    }
}

File: Templates/C#/SolutionTemplate/Project/Db/InMemoryDatabase.cs
using System.Collections.Generic;
using System.Linq;
using Project.Models;

namespace Project.Db
{
    public class InMemoryDatabase : IDatabase
    {
        private readonly Dictionary<int, Student> students = new Dictionary<int, Student>();

        public Student CreateStudent(Student student)
        {
            int nextId = students.Any() ? students.Keys.Max() + 1 : 0;
            student.Id = nextId;
            students[student.Id] = student;
            return student;
        }

        public Student GetStudent(int id)
        {
            students.TryGetValue(id, out var student);
            return student;
        }

        public void UpdateStudent(Student student)
        {
            students[student.Id] = student;
        }

        public void DeleteStudent(int id)
        {
            students.Remove(id);
        }

        public List<Student> GetAllStudents()
        {
            return students.Values.ToList();
        }
    }
}
